[IMP] remove ci-tools dependency

PierrickBrun opened the pull request:
since postgres 13, dropdb can be --forced, this removes the need for a ci-drop-db

about the keepdb, a separate job with a rule makes it works

about the db creation:
 - ci-getdevdb is just a createdb (with `|| true` to not fail if it already exists)
 - ci-getdb can be replaced by a createdb -T (there is no spare anymore but I'm not convinced this feature is that important)

PierrickBrun commented on .gitlab-ci.yml.jinja:
```diff
@@ -81,16 +78,29 @@ test:
         path: shared/coverage.xml
 
+# Drop the database to restart from a fresh one
+drop_database:
```
because I want to run this job only if there is no keepdb.

If I run a pipeline with `keepdb` while having no database it would not be created (if I drop & create the db in this job).

Then the migrate will fail.

I could create a separate job for database creation but that would create another stage that is not necessary

hparfr replied:
this kind of aglo:

test if db exists
if no: 
  createdevdb
  exit Success
if keepdb:
  print "remove keepdb before refresh. Nothing done."
  exit Failure

dropdb

if emptydb (new label ?)
  createdevdb
else
  createdb
exit Success

PierrickBrun commented on .gitlab-ci.yml.jinja:
```diff
@@ -81,16 +78,29 @@ test:
-    - ci-getdevdb
-    - docker-compose run odoo odoo --stop-after-init
+    # Replace this lines with the following ones if you want to use a template database
+    # - createdb $BUILD_NAME -T ${BUILD_NAME}_template || true
+    # - docker-compose run odoo click-odoo-update
```
yes, ci-getdb is different than a -T but I don't see why it would be a problem:

 - ci-getdb fails often and does not give any indication about the origin of the failure
 - ci-getdb relies on cidbservice (which needs an additional key, while createdb only needs pg access)
 - createdb -T has understandable its errors can be fixed by anyone
 - let's be honest, the refactoring of getdb/cidbservice will not happen in at least 6 month

I'm not convinced the spare database feature is worth all this

sebastienbeau replied:
I agree to move to createdb -T.
Then I will propose a new PR that will introduce a basic spare feature
- stupid python script that rename a spare or create a db from template if there is not spare
- a job in the CI that generate the spare

This spare feature will be an option of copier (as only big db project will need)

PierrickBrun replied:
good idea to use a CI job IMO. 
For instance it can create x spares in the morning and then 1 at the end of each pipeline

PierrickBrun commented:
@sebastienbeau @hparfr 

I've refactored this MR, tested on the ugess project (which is now updated via copier)

I've added a `emptydb` tag

It works well with & without keepdb and with & without emptydb.

Since `only:` clashes with `rules:` I've added a workflow (https://docs.gitlab.com/ee/ci/yaml/#workflow ) at the top. It is the same as `only:` but on the whole pipeline.

hparfr commented on .gitlab-ci.yml.jinja:
```diff
@@ -47,53 +58,83 @@ before_script:
+  script:
+    - docker-compose kill
+    - dropdb --force --if-exists $BUILD_NAME
+    - createdb $BUILD_NAME -T ${CI_PROJECT_NAME}_template || true
```
why `|| true` ?

PierrickBrun replied:
I think it was needed before but it isn't now, I'll remove it

hparfr commented on .gitlab-ci.yml.jinja:
```diff
@@ -47,53 +58,83 @@ before_script:
+    # if emptydb tag is set this job will not run
+    - if: $CI_MERGE_REQUEST_LABELS !~ /.*emptydb.*/
+    # if keepdb is set, this job will run
+    - if: $CI_MERGE_REQUEST_LABELS =~ /.*keepdb.*/
```
why the `/.*` syntax ? not the same as skiptest ?

PierrickBrun replied:
I did not notice the difference, indeed the `.*` is not useful. I'll remove it

hparfr commented on .gitlab-ci.yml.jinja:
```diff
@@ -47,53 +58,83 @@ before_script:
+  stage: migrate
+  script:
+    - export PGDATABASE=$BUILD_NAME
+    - echo "DB_NAME=$BUILD_NAME" >> .env
```
no alignment between PGDATABASE and DB_NAME ?

PierrickBrun replied:
I did not look into it (this is the same code as before)

I think this is something to look into in another PR
